fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownAssertOneRowProjectRule.java

```diff
@@ -14,7 +14,7 @@
 public class PushDownAssertOneRowProjectRule extends TransformationRule {
     public PushDownAssertOneRowProjectRule() {
         super(RuleType.TF_PUSH_DOWN_ASSERT_ONE_ROW_PROJECT, Pattern.create(OperatorType.LOGICAL_ASSERT_ONE_ROW)
-                .addChildren(Pattern.create(OperatorType.LOGICAL_PROJECT, OperatorType.PATTERN_LEAF)));
+                .addChildren(Pattern.create(OperatorType.LOGICAL_PROJECT, OperatorType.LOGICAL_AGGR)));
```

Comment:
why could only be LOGICAL_AGGR?

Reply:
Because one of the main optimizations of AssertOneRow is that when AssertOneRow is on an aggregate node, AssertOneRow may be optimized away. The specific optimization rules are in PruneAssertOneRowRule. In other cases, there is no need to push down AssertOneRow.